File: AdisyonUygulamaJavaXMLsiz/app/src/main/java/com/example/adisyonuygulamajava/services/retrofit/RetrofitClient.java
package com.example.adisyonuygulamajava.services.retrofit;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {

    private RetrofitClient() {}

    public static Retrofit getRetrofitClient(String baseUrl) {
        return new Retrofit.Builder()
                .baseUrl(baseUrl)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
    }
}

File: AdisyonUygulamaJetpack/app/src/main/java/com/example/adisyonuygulamajetpack/services/retrofit/RetrofitClient.kt
package com.example.adisyonuygulamakotlin.services.retrofit

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitClient {
    companion object {
        fun getRetrofitClient(baseUrl:String):Retrofit{
            return Retrofit.Builder().baseUrl(baseUrl).addConverterFactory(GsonConverterFactory.create()).build()
        }
    }

}